runtime/run.go

```diff
@@ -33,44 +33,29 @@ func Exec(ctx context.Context, log logr.Logger, runtime *Runtime, transform Tran
 
 // printFunctionIO prints the whole FunctionIO to stdout, so Crossplane can
 // pick it up again.
-func printFunctionIO(iof *xfnv1alpha1.FunctionIO, log logr.Logger) error {
+func printFunctionIO(iof *xfnv1alpha1.FunctionIO, log logr.Logger) ([]byte, error) {
 	log.V(1).Info("Marshalling FunctionIO")
 	fnc, err := yaml.Marshal(iof)
 	if err != nil {
-		return fmt.Errorf("failed to marshal function io: %w", err)
+		return []byte{}, fmt.Errorf("failed to marshal function io: %w", err)
 	}
-
-	fmt.Println(string(fnc))
-	return nil
+	return fnc, nil
 }
 
-func RunCommand(ctx *cli.Context, transforms []Transform) error {
-	log := logr.FromContextOrDiscard(ctx.Context)
+func RunCommand(ctx *context.Context, input []byte, transforms []Transform) ([]byte, error) {
```

Comment:
It's very weird to take a pointer to `context.Context` and usually not at all what you want

```suggestion
func RunCommand(ctx context.Context, input []byte, transforms []Transform) ([]byte, error) {
```